luthier/src/main/java/com/yahoo/bard/webservice/config/luthier/factories/SingleDataSourcePhysicalTableFactory.java

```diff
@@ -3,30 +3,37 @@
 package com.yahoo.bard.webservice.config.luthier.factories;
 
 import com.yahoo.bard.webservice.config.luthier.Factory;
+import com.yahoo.bard.webservice.config.luthier.LuthierFactoryException;
 import com.yahoo.bard.webservice.config.luthier.LuthierIndustrialPark;
 import com.yahoo.bard.webservice.config.luthier.LuthierValidationUtils;
 import com.yahoo.bard.webservice.data.config.LuthierPhysicalTableParams;
 import com.yahoo.bard.webservice.data.config.LuthierTableName;
 import com.yahoo.bard.webservice.data.dimension.DimensionColumn;
 import com.yahoo.bard.webservice.data.metric.MetricColumn;
-import com.yahoo.bard.webservice.data.time.DefaultTimeGrain;
+import com.yahoo.bard.webservice.data.time.Granularity;
+import com.yahoo.bard.webservice.data.time.ZonedTimeGrain;
 import com.yahoo.bard.webservice.table.ConfigPhysicalTable;
 
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.node.ObjectNode;
 
+import com.yahoo.bard.webservice.util.GranularityParseException;
 import org.joda.time.DateTimeZone;
 
 import java.util.LinkedHashMap;
 import java.util.LinkedHashSet;
-import java.util.Locale;
 
 /**
  * A factory that is used by default to support Simple (non-Composite) Physical Table.
  */
-abstract class SingleDataSourcePhysicalTableFactory implements Factory<ConfigPhysicalTable> {
+abstract public class SingleDataSourcePhysicalTableFactory implements Factory<ConfigPhysicalTable> {
```

Comment:
We typically put `public` before `abstract`: `public abstract class SingleDataSource...`

Reply:
done